File: mobile-engage/src/main/java/com/emarsys/mobileengage/event/EventServiceInternal.java
package com.emarsys.mobileengage.event;

import com.emarsys.core.api.result.CompletionListener;

import java.util.Map;

public interface EventServiceInternal {
    String trackCustomEvent(
            String eventName,
            Map<String, String> eventAttributes,
            CompletionListener completionListener);

    void trackCustomEventAsync(
            String eventName,
            Map<String, String> eventAttributes,
            CompletionListener completionListener);

    String trackInternalCustomEvent(
            String eventName,
            Map<String, String> eventAttributes,
            CompletionListener completionListener);

    void trackInternalCustomEventAsync(
            String eventName,
            Map<String, String> eventAttributes,
            CompletionListener completionListener);
}

File: mobile-engage/src/main/java/com/emarsys/mobileengage/event/EventServiceInternal.kt
package com.emarsys.mobileengage.event

import com.emarsys.core.api.result.CompletionListener

interface EventServiceInternal {
    fun trackCustomEvent(
        eventName: String,
        eventAttributes: Map<String, String>?,
        completionListener: CompletionListener?
    ): String?

    fun trackCustomEventAsync(
        eventName: String,
        eventAttributes: Map<String, String>?,
        completionListener: CompletionListener?
    )

    fun trackInternalCustomEvent(
        eventName: String,
        eventAttributes: Map<String, String>?,
        completionListener: CompletionListener?
    ): String?

    fun trackInternalCustomEventAsync(
        eventName: String,
        eventAttributes: Map<String, String>?,
        completionListener: CompletionListener?
    )
}